compute_tools/src/spec.rs

```diff
@@ -511,17 +511,21 @@ pub fn handle_grants(spec: &ComputeSpec, connstr: &str, client: &mut Client) ->
         .map(|r| r.name.pg_quote())
         .collect::<Vec<_>>();
 
-    for db in &spec.cluster.databases {
-        let dbname = &db.name;
+    let is_webaccess = spec.cluster.roles.iter().any(|r| r.name == "web_access");
 
-        let query: String = format!(
-            "GRANT CREATE ON DATABASE {} TO {}",
-            dbname.pg_quote(),
-            roles.join(", ")
-        );
-        info!("grant query {}", &query);
+    if is_webaccess {
+        for db in &spec.cluster.databases {
+            let dbname = &db.name;
 
-        client.execute(query.as_str(), &[])?;
+            let query: String = format!(
+                "GRANT CREATE ON DATABASE {} TO {}",
```

Comment:
This grant is needed to create extensions cc @lubennikovaav 

So we should grant it to every role on every db in some way or another

Reply:
if we will not do it automatically, the DX will be worse than it was

Reply:
But we want to maintain the mantra that we "ARE postgres in the cloud", i.e. everything is the same

Reply:
It's harder for users to accept when you take sth back.

Initially, they were able to create extensions w/o thinking about roles. In addition, for users who are new to Postgres and heard about PostGIS, it'll be much harder to start experimenting.

Reply:
>But we want to maintain the mantra that we "ARE postgres in the cloud", i.e. everything is the same

We are in terms of features and compatibility.
But we do not provide superuser access, which leads to some limitations and we must provide workarounds.

Currently, users cannot execute GRANT/REVOKE on their databases and some other objects.
Can't remember why exactly, most likely because it requires superuser.

So we agreed to adjust all roles privileges and always start with "every user can access everything". Because we believe that it fits "beginner" users better. This was not only for `web_access`, IIRC.

Maybe it is time to change our grant model, but I would like to see some design discussion, before we do this.


I still do not quite understand what problem this PR aims to fix.
@save-buffer , can you please adjust description / send link to thread?

Reply:
Adjusted description. Thread: https://neondb.slack.com/archives/C03QLRH7PPD/p1685647864023039

Reply:
This thread is now outdated with the introduction of `neon_superuser`, which will be granted all priveleges to all databases.